internal/compiler-interface/src/main/contraband/other.json

```diff
@@ -10,6 +10,7 @@
         { "name": "name",        "type": "String"          },
         { "name": "api",         "type": "lazy Companions" },
         { "name": "apiHash",     "type": "int"             },
+        { "name": "apiHashExtra","type": "int"             },
```

Comment:
`apiHash` means hash of API info (sum total of public signatures).
`apiHashExtra` doesn't make sense. I would suggest something like `extraHash`, and also let's add `"docs"` to explain subtle differences of those like in https://github.com/sbt/zinc/blob/1.x/internal/compiler-interface/src/main/contraband/incremental.json

Reply:
Also I assume we want to keep binary compatibility of compiler interface, so it might be good to use `"since"` feature so we can generate bincompat overloads.